cmd/soroban-cli/src/commands/events.rs

```diff
@@ -159,12 +159,12 @@ impl Cmd {
             }
         }
 
+        let network = self.network.get(&self.locator)?;
+
         // Validate contract_ids
-        for id in &mut self.contract_ids {
-            utils::contract_id_from_str(id).map_err(|e| Error::InvalidContractId {
-                contract_id: id.clone(),
-                error: e,
-            })?;
+        for id in &self.contract_ids {
+            self.locator
+                .resolve_contract_id(id, &network.network_passphrase)?;
```

Comment:
I haven't read this file in detail, but this wouldn't be needed if we used the strkey type instead then below you could just to `to_string` on it.